ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3BaseHandler.java

```diff
@@ -135,4 +148,70 @@ public static boolean isMultipartListPartRequest(RestRequest restRequest) {
     return restRequest.getRestMethod() == RestMethod.GET && restRequest.getArgs().containsKey(S3_REQUEST)
         && restRequest.getArgs().containsKey(UPLOAD_ID_QUERY_PARAM);
   }
+
+  protected class DefaultCallbackChain {
```

Comment:
Can we use a different name indicating that it's erroring out.

Reply:
Sure. Let me think about the better name for it. May change at the next PR.
For now, I'll merge it in since Sophie will start the list based on the PR.